File: Anagrams.cpp
class Solution {
public:    
    /**
     * @param strs: A list of strings
     * @return: A list of strings
     */
    vector<string> anagrams(vector<string> &strs) {
        // write your code here
        unordered_map<string, int> map;
        for (auto s : strs){
            sort(s.begin(), s.end());
            ++map[s];
        }
        vector<string> ret;
        for (const auto& s : strs){
            string sorted_s(s);
            sort(sorted_s.begin(), sorted_s.end());
            if (map[sorted_s] > 1)
                ret.emplace_back(s);
        }
        return ret;
    }
};

File: Anagrams.py
class Solution:
    # @param strs: A list of strings
    # @return: A list of strings
    def anagrams(self, strs):
        # write your code here
        def add(hashstr, s):
            ss = list(s)
            ss.sort()
            ns = ''.join(ss)
            ns.strip()
            hashstr[ns].add(s)
            
            
        hashstr = collections.defaultdict(set)
        for s in strs:
            add(hashstr, s)
        
        ans = []
        for k, v in hashstr.items():
            if len(v) > 1:
                ans.extend(v)
        return ans
            